From a9d5efb12096b47e063f53ee381d043b580b061e Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Tue, 5 Jul 2016 11:38:36 +0300
Subject: [PATCH 0335/1345] kernel-include: mv_nss_metadata change commentary

commit  abd6c1f34fcf8ad780ee803e5a9dad08a9a4a7d5 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Change commentaries in include/net/gnss/mv_nss_metadata.h

Change-Id: Id72fccabf44b438c76b4c17d2ee025604234d95f
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30945
Reviewed-by: Lior Amsalem <alior@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 include/net/gnss/mv_nss_metadata.h |   59 +++++-------------------------------
 1 file changed, 7 insertions(+), 52 deletions(-)

diff --git a/include/net/gnss/mv_nss_metadata.h b/include/net/gnss/mv_nss_metadata.h
index 9949fa3..5cf9596 100644
--- a/include/net/gnss/mv_nss_metadata.h
+++ b/include/net/gnss/mv_nss_metadata.h
@@ -81,7 +81,7 @@
 *                         7 = ARP
 *       cos         - Class of service.
 *       reason      - Packet origination context.
-*       opaque      - Application opaque handle.
+*       opaque      - Opaque data.
 *       reserved    - Reserved fields.
 */
 struct mv_nss_metadata {
@@ -105,57 +105,12 @@ struct mv_nss_metadata {
 *       port_src    - Source virtual port.
 *       port_dst    - Destination virtual port.
 *       type        - Packet information.
-*                       bits [0-2]: Frame type:
-*                         0 - Standard Ethernet II frame
-*                         1 - Proprietary A-MSDU short frame
-*                         2 - Proprietary A-MSDU basic frame
-*                       bit [3] - Validity condition:
-*                         0 - bits [4-31] are invalid
-*                         1 - bits [4-31] are valid
-*                       bits [8-14] - L3 offset: IPv4 or IPv6 offset in the packet.
-*                          This field indicates the beginning of Layer3 (in octets).
-*                       bit [15] - MAC to me: MAC DA lookup results
-*                         0 - MAC DA is unknown (not MAC to me)
-*                         1 - MAC DA is known unicast or multicast MAC address
-*                             (MAC to me)
-*                       bits [16-20] - IP Header length: IP Header Length in
-*                            4-octet words.
-*                            IPv4: IPv4 header length including options
-*                            if exist
-*                            IPv6: IPv6 header length including extension
-*                            header if exists
-*                       bits [21-23] - L4 info: L4 parsing results
-*                         0 - Unknown
-*                         1 - TCP
-*                         2 - TCP + checksum error
-*                         3 - UDP
-*                         4 - UDP lite
-*                         5 - UDP + checksum error
-*                         6 - IGMP
-*                         7 - Other
-*                       bits [24-25]: VLAN info - Number of VLANs in the packet
-*                         0 - Untagged
-*                         1 - Single Tag
-*                         2 - Double tag
-*                         3 - Reserved
-*                       bit [26] - Management or non-management packet type
-*                         0 - Non-management packet
-*                         1 - Management packet
-*                       bits [27-28] - L2 info: indicates type of L2 packet
-*                         0 - Unicast
-*                         1 - Multicast
-*                         2 - IP Multicast
-*                         3 - Broadcast
-*                       bits [29-31] - L3 info: L3 parsing results
-*                         0 - Unknown
-*                         1 - IPv4
-*                         2 - IPv4 fragment
-*                         3 - IPv4 with options
-*                         4 = IPv4 with errors (checksum, TTL, etc)
-*                         5 = IPv6
-*                         6 = IPv6 with extension(s) header
-*                         7 = ARP
-*       opaque      - A-MSDU opaque application handle from A-MSDU context.
+*                       bit [0] - Validity condition.
+*                         1 - bits [4-31] are invalid
+*                       bits [1]: Frame type:
+*                         0 - Standard Ethernet II frame.
+*                         1 - Proprietary A-MSDU frame.
+*       opaque      - A-MSDU application handle from A-MSDU context.
 *       cos         - Class of service.
 *       reason      - Packet origination context.
 *       reserved1   - Reserved fields.
-- 
1.7.9.5

